### PR TITLE
[V1.1][ISSUE-188] Model Adapter code refine and bug fix

### PR DESCRIPTION
**Pull request by XinyaoWa**
- Refine requirements.txt and can install in vanilla env 
- remove optuna 
- remove useless code (draft / model_bak)
- disable distiller save logits in distribution mode
- bug fix
    -  distiller can't save logits only after 1 epoch, add set_epoch to fix
    -  Trainer will fail to show parameter in tensorboard if directly call evaluate function, add flag to fix
    -  In start_e2eaiok_docker.py, linux_distribution attribute is deprecated in python 3.8, fix by add distro option

### CODE REVIEW

**Comment by XinyaoWa** [comment]
@xuechendi @zigzagcai @zhouyu5 Please have a code review, thanks!

**Comment by xuechendi** [comment]
@XinyaoWa , other parts looks good to me, one question is what draft folder is used for?

**Comment by XinyaoWa** [comment]
> @XinyaoWa , other parts looks good to me, one question is what draft folder is used for?

Before it's used to keep backup code, but all the codes are not needed in current MA. Thus I remove it in public repo and only keep it in my local.

**Comment by xuechendi** [comment]
> > @XinyaoWa , other parts looks good to me, one question is what draft folder is used for?
> 
> Before it's used to keep backup code, but all the codes are not needed in current MA. Thus I remove it in public repo and only keep it in my local.

I see, thanks for the clarify, approved here

**Review by xuechendi — APPROVED** [review]
approved

**Comment by xuechendi** [comment]
@XinyaoWa , do you expect another reviewer to review this PR as well? If you yes, I'll hold this PR's merge until we received another approval or pls fix the conflict and I'll merge this one

**Comment by XinyaoWa** [comment]
> @XinyaoWa , do you expect another reviewer to review this PR as well? If you yes, I'll hold this PR's merge until we received another approval or pls fix the conflict and I'll merge this one

Yes, I have asked zheng and yu for review, please wait some time, thanks!

**Comment by zigzagcai** [comment]
LGTM

**Review by zigzagcai — APPROVED** [review]

**Review by zhouyu5 — APPROVED** [review]

**Comment by XinyaoWa** [comment]
> > @XinyaoWa , do you expect another reviewer to review this PR as well? If you yes, I'll hold this PR's merge until we received another approval or pls fix the conflict and I'll merge this one
> 
> Yes, I have asked zheng and yu for review, please wait some time, thanks!

@xuechendi  Now it's able to merge, thanks!
